homeassistant/components/smappee/binary_sensor.py

```diff
@@ -48,54 +65,33 @@ async def async_setup_entry(
 class SmappeePresence(BinarySensorEntity):
     """Implementation of a Smappee presence binary sensor."""
 
+    _attr_device_class = BinarySensorDeviceClass.PRESENCE
+
     def __init__(self, smappee_base, service_location):
         """Initialize the Smappee sensor."""
         self._smappee_base = smappee_base
         self._service_location = service_location
-        self._state = self._service_location.is_present
-
-    @property
-    def name(self):
-        """Return the name of the binary sensor."""
-        return f"{self._service_location.service_location_name} - {PRESENCE_PREFIX}"
-
-    @property
-    def is_on(self):
-        """Return if the binary sensor is turned on."""
-        return self._state
-
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return BinarySensorDeviceClass.PRESENCE
-
-    @property
-    def unique_id(
-        self,
-    ):
-        """Return the unique ID for this binary sensor."""
-        return (
-            f"{self._service_location.device_serial_number}-"
-            f"{self._service_location.service_location_id}-"
+        self._attr_name = (
```

Comment:
```suggestion
        self._attr_is_on = service_location.is_present
        self._attr_name = (
```

We should leave setting the state to an initial value, or do I miss something?

Reply:
I noticed that `async_add_entities(..., True)` update before add was True, so I interpreted it as, before you're added, the sensors update method will be called, and since that does the same, it would mean the same.

Reply:
`async_update` gets called before the entity is added to the platform when `updated_before_add=True`

https://github.com/home-assistant/core/blob/70c6bceaee828cedf7f5328b2de92828e2dc312f/homeassistant/helpers/entity_platform.py#L573